docs/src/tutorials/conic/optimal-power-flow.jl

```diff
@@ -0,0 +1,286 @@
+# See the [Complex number support](@ref) section of the manual for more details.
+
+# For another example of modeling with complex decision variables
+# see the [Quantum state discrimination](@ref) tutorial.
+
+# ## Required packages
+
+# This tutorial requires the following packages:
+
+using JuMP
+import Ipopt
+import SCS
+import LinearAlgebra: diag, diagm
+import SparseArrays: sparse, spdiagm, sparsevec
+import DataFrames: DataFrame, rename
+import Test
+
+# ## Formulation
+
+# Optimal power flow problems for electrical transmission typically pose the
+# following question: what is the most cost-effective operation of electricity
+# generators while meeting constraints on the safe limits of network components?
+
+# We will use the 9 _bus_ network test case `case9mod` to explore this problem
+# following the example of Krasko and Rebennack (2017).
+
+# Here *bus* and network *node* are taken as analogous terms, as are *branch* and transmission *line*.
+
+# The graph of the network shows three nodes each used for the different purposes of
+# generation ``G`` (nodes 1, 2 and 3),
+# trans-shipment (nodes 4, 6 and 8) 
+# and demand ``L`` (5, 7, and 9).
+
+# ![Nine Nodes](../../assets/case9mod.png)
+
+# For future reference, let's name the number of nodes in the network:
+N = 9
+
+# The network data can be summarised as follows.
+
+# Generation power bounds (active and reactive):
+P_G_lb = sparsevec([1, 2, 3], [10, 10, 10], N)
+P_G_ub = sparsevec([1, 2, 3], [250, 300, 270], N)
+
+Q_G_lb = sparsevec([1, 2, 3], [-5, -5, -5], N)
+Q_G_ub = sparsevec([1, 2, 3], [300, 300, 300], N)
+
+# Power demand levels (active, reactive and complex form):
+P_D_fx = sparsevec([5, 7, 9], [54, 60, 75], N)
+Q_D_fx = sparsevec([5, 7, 9], [18, 21, 30], N)
+
+S_D_fx = P_D_fx + im * Q_D_fx
+
+# The key decision variables here are the real power injections ``P^G`` and
+# reactive power injections ``Q^G``over the allowed range of the generators.
+# All other buses must restrict their generation variables to 0.
+# On the other hand these non-generator nodes have a fixed 
+# real and reactive power demand, denoted ``P^L`` and ``Q^L`` respectively
+# (fixed at 0 in the case of trans-shipment and generator nodes).
+
+# The cost of operating each generator is here modeled as a quadratic function of its real power output;
+# in our specific test case, the objective function to minimize is
+# ```math
+# \begin{align}
+#     \min \;\; & \; 0.11 \;\; (P^G_1)^2 +   5 P^G_1 + 150  \\
+#        + \;\; & 0.085 \; (P^G_2)^2 + 1.2 P^G_2 + 600  \\
+#          \;\; & 0.1225 \;  (P^G_3)^2 +     P^G_3 + 335 \\
+# \end{align}
+# ```
+# Let's create a basic JuMP model with these settings:
+model = Model(Ipopt.Optimizer)
+set_silent(model)
+
+@variable(model, P_G_lb[i] <= P_G[i = 1:N] <= P_G_ub[i])
+
+#! format: off 
+@objective(model, Min,
+      0.11*P_G[1]^2 +   5*P_G[1] + 150
++    0.085*P_G[2]^2 + 1.2*P_G[2] + 600
++   0.1225*P_G[3]^2 +     P_G[3] + 335)
+#! format: off 
```

Comment:
Can we not just let the formatter do its thing?

Reply:
Ok! 😃